webClient/src/app/editor/code-editor/code-editor.component.ts

```diff
@@ -294,7 +294,24 @@ export class CodeEditorComponent implements OnInit, OnDestroy {
       let nextFileContext = this.editorControl.fetchActiveFile();
       this.selectFile(nextFileContext, true);
     } else {
-      this.codeEditorService.closeFile(fileContext);
+      if(fileContext.changed) {
+        let directory = fileContext.model.path || this.editorControl.activeDirectory;
+        const title = 'Do you want to save the changes you made to  ' + fileContext.name + '\'?';
+        const warningMessage = 'Your changes will be lost if you don\'t save them.';
+        let response = this.monacoService.confirmAction(title, warningMessage).subscribe(response => {
```

Comment:
Why are we duplicating the above lines in `webClient/src/app/core/menu-bar/menu-bar.component.ts` ? Can't we call `promptToSave `from `menu-bar.component.ts`? Then call `this.codeEditorService.closeFile(fileContext)` based on resolve/reject

or vice-versa (create method in `code-editor.component.ts` and then call it from `menu-bar` whichever is easier)